### ISSUE TITLE
Channel Tooltips

### ISSUE THREAD

**Issue by prateeksriv** [mattermost/platform#6346]
For feature requests, please see: http://www.mattermost.org/feature-requests/.
For troubleshooting questions, please post on the forums: http://forum.mattermost.org/.

For issues with the product, please follow the steps at http://www.mattermost.org/filing-issues/ and post here in the following format:

#### Summary
When browsing channels, the channel name is displayed but the mousover (hover) action shows nothing. I prefer to see channel header to make a decision if it is worth joining or not.

#### Steps to reproduce
Hover over the channel name in the navigation bar on the left.

#### Expected behavior
Tool-tip should show the channel header.

#### Observed behavior
No tool-tip

#### Possible fixes
Here is the code change that I have made and tested manually.
Please help me push it upstream.
https://github.com/prateeksriv/platform/tree/PLT-6346

**Comment by jasonblais** [comment]
Thanks @prateeksriv, found your ticket 👍 

To confirm, you're adding the channel header when hovering over a channel on your left-hand sidebar? I think you may have joined the channel already though?

When you're joining a new channel, we show the channel purpose on the list. Wondering if that would help in making a decision whether to join a channel or not?

![image](https://cloud.githubusercontent.com/assets/13119842/25781078/34e2dca2-3302-11e7-94a4-dee7a10eee57.png)

**Comment by prateeksriv** [comment]
Hi @jasonblais ,

Thank you for inputs. Yes, I guess, for joining the information is sufficient. But if you are a member of many channels, the sidebar can be quite long and you may want to know the channel details before you click into one of them.

In my organization, we use Mattermost a little differently. We have an integration with the ticketing system so that the teams can discuss in context of the ticket number. 
The integration script creates a channel with the name of channel and channel header is the customer name.
So, if I am browsing the long list, I do not want to click the channel ( marking it read as side-effect) unless I know it is for the customer I am interested in. I found it useful.

**Comment by jasonblais** [comment]
Got it, thanks @prateeksriv 

Adding a tooltip that displays either the channel header or purpose is something we can consider adding. I could see that useful in other cases as well. For the general user base, a lot of people use Markdown links in the channel header, so it could be hard to parse that in the tooltip.

I'm wondering for your use case though, would it be easier if the integration script created the channel name as `ticket XXX: customer name`? That would be an even faster way to know if it's a customer you care about and wouldn't force you to hover over the channel name.

**Comment by prateeksriv** [comment]
The channel name was limited in length. I don't remember right now.  The ticket id takes 10 chars and then very few chars for names.

**Comment by prateeksriv** [comment]
Hi @jasonblais, next steps for me?

**Comment by jasonblais** [comment]
Hey @prateeksriv, sorry for the delay. I've been travelling so just catching up on things.

I'd propose considering to show the Purpose on channel tooltip instead of the header (you can use the integration to supply the company name for header and purpose)

The header is typically supplied with information such as links, rather than to describe the purpose of the channel. It also supports Markdown-formatting, so parsing it to the tooltip will be tricky.

The purpose is meant to describe what the channel is about, and will be useful for other Mattermost deployments.

Let me know what you think? I can queue it for proposal to our UX design team.

**Comment by jasonblais** [comment]
Thanks again @prateeksriv, really appreciate your thoughtfulness and effort for contributing the pull request.

Would love to hear your thoughts on the above note, open for discussion.

**Comment by lindy65** [comment]
Hi @prateeksriv, we haven't received any feedback so will close the issue off for now.

If you'd still like to continue the discussion, feel free to reopen this issue.
